lightning/src/chain/keysinterface.rs

```diff
@@ -222,9 +229,17 @@ pub trait BaseSign {
 	/// Create a signature for a counterparty's commitment transaction and associated HTLC transactions.
 	///
 	/// Note that if signing fails or is rejected, the channel will be force-closed.
+	///
+	/// Policy checks should be implemented in this function, including checking the amount
+	/// sent to us and checking the HTLCs.
 	//
 	// TODO: Document the things someone using this interface should enforce before signing.
 	fn sign_counterparty_commitment(&self, commitment_tx: &CommitmentTransaction, secp_ctx: &Secp256k1<secp256k1::All>) -> Result<(Signature, Vec<Signature>), ()>;
+	/// Validate the counterparty's revocation.
+	///
+	/// This is required in order for the signer to make sure that the state has moved
+	/// forward and it is safe to sign the next counterparty commitment.
+	fn validate_counterparty_revocation(&self, idx: u64, secret: &SecretKey);
```

Comment:
Shouldn't this return something with an `Err` variant, same as above.

Reply:
Currently all validation in `BaseSign` panics rather than returns an Err.  See for example `EnforcingSigner::sign_counterparty_commitment` calling `verify_counterparty_commitment_tx`.  I think we should have an Err strategy, but should probably implement that everywhere in `BaseSign` / `EnforcingSigner`, rather than piecemeal.

Reply:
> Currently all validation in BaseSign panics rather than returns an Err. See for example EnforcingSigner::sign_counterparty_commitment

There are two completely separate things here, though - `EnforcingSigner` panics because its a test utility that should immediately fail the test if we hit an error. `BaseSign`, the interface, does not panic, because implementors of it can chose whether to panic or force-close the channel on failure. I dont think any existing functions require the implementor to panic?

Reply:
OK, good point.  But we also need this change on the `validate_holder_commitment` fn.

Please take a look at the new commit.